src/CMakeLists.txt

```diff
@@ -0,0 +1,113 @@
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1y -Wall -Wextra -rdynamic -g3 -Wno-deprecated -fsanitize=address")
+
+find_package(PkgConfig)
+pkg_check_modules(GTKMM gtkmm-3.0)
+pkg_check_modules(LIBNOTIFYMM libnotifymm-1.0)
+pkg_check_modules(LIBCANBERRA libcanberra)
+pkg_check_modules(GSTREAMERMM gstreamermm-1.0)
+link_directories(
+    ${GTKMM_LIBRARY_DIRS} ${LIBNOTIFYMM_LIBRARY_DIRS} ${LIBCANBERRA_LIBRARY_DIRS} ${GSTREAMERMM_LIBRARY_DIRS})
+include_directories(
+    ${GTKMM_INCLUDE_DIRS} ${LIBNOTIFYMM_INCLUDE_DIRS} ${LIBCANBERRA_INCLUDE_DIRS} ${GSTREAMERMM_INCLUDE_DIRS})
+
+set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
+
+find_package(Tox)
+include_directories(${TOX_INCLUDE_DIR})
+
+find_package(Flatbuffers)
+include_directories(${FLATBUFFERS_INCLUDE_DIR})
+
+add_subdirectory(tox)
+add_subdirectory(i18n)
+add_subdirectory(resources/ui)
+add_subdirectory(resources/style)
+add_subdirectory(resources/icon)
+add_subdirectory(resources/flatbuffers)
+
+set(SOURCES
+    main.cpp
+    gTox.cpp
+    config.cpp
+
+    dialog/error.cpp
+    dialog/profile_selection.cpp
+    dialog/profile_create.cpp
+    dialog/main.cpp
+    dialog/chat.cpp
+    dialog/settings.cpp
+
+    widget/avatar.cpp
+    widget/contact.cpp
+    widget/main_menu.cpp
+    widget/label.cpp
+    widget/chat_input.cpp
+    widget/chat_message.cpp
+    widget/chat_action.cpp
+    widget/chat_bubble.cpp
+
+    utils/builder.cpp
+    utils/dispatcher.h
+)
+SET_SOURCE_FILES_PROPERTIES(${GRESOURCE} PROPERTIES GENERATED 1)
+add_executable(${PROJECT_NAME}
+    ${SOURCES}
+    ${GRESOURCE}
+)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+target_link_libraries(${PROJECT_NAME} ${GTKMM_LIBRARIES} ${LIBNOTIFYMM_LIBRARIES} ${LIBCANBERRA_LIBRARIES} ${GSTREAMERMM_LIBRARIES} ${TOX_LIBRARY} -lpthread -lsqlite3 Toxmm)
+add_dependencies(${PROJECT_NAME}
+     gtox-style-resource
+     gtox-icon-resource
+     gtox-ui-resource
+     gtox-flatbuffers-resource
+)
+install(PROGRAMS "${PROJECT_BINARY_DIR}/${PROJECT_NAME}"
+        DESTINATION bin
+        PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ GROUP_EXECUTE GROUP_READ WORLD_EXECUTE WORLD_READ)
+
+#GENRATE POT FILES
+find_program(GETTEXT_XGETTEXT_EXECUTEABLE xgettext)
+if(NOT GETTEXT_XGETTEXT_EXECUTEABLE)
+    message(FATAL_ERROR "xgettext not found")
+endif()
+
+find_program(GETTEXT_MSGCAT_EXECUTEABLE msgcat)
+if(NOT GETTEXT_MSGCAT_EXECUTEABLE)
+    message(FATAL_ERROR "msgcat not found")
+endif()
+
+FUNCTION(PREPEND var prefix)
+   SET(listVar "")
+   FOREACH(f ${ARGN})
+      LIST(APPEND listVar "${prefix}/${f}")
+   ENDFOREACH(f)
+   SET(${var} "${listVar}" PARENT_SCOPE)
+ENDFUNCTION(PREPEND)
+
+PREPEND(LAYOUT_P "resources/ui" ${LAYOUT})
+add_custom_command(
+   COMMENT "Generate pot file"
+   TARGET ${PROJECT_NAME}
+   PRE_BUILD
+   COMMAND ${GETTEXT_XGETTEXT_EXECUTEABLE} -o "${CMAKE_CURRENT_SOURCE_DIR}/i18n/source.pot" --add-location=file --from-code=UTF-8 --sort-output --c++ --omit-header -k_ -s ${SOURCES}
```

Comment:
If I remember correctly...
because the following merge command didn't work right without sorting.